pkg/secrets/plugins/filesystem/store.go

```diff
@@ -1,138 +1,132 @@
 package filesystem
 
 import (
+	"context"
+	"fmt"
 	"os"
 	"path/filepath"
 
 	"get.porter.sh/porter/pkg/config"
 	"get.porter.sh/porter/pkg/secrets"
 	"get.porter.sh/porter/pkg/secrets/plugins"
 	"get.porter.sh/porter/pkg/secrets/plugins/host"
-	"github.com/carolynvs/aferox"
-	cnabsecrets "github.com/cnabio/cnab-go/secrets"
-	"github.com/hashicorp/go-hclog"
+	"get.porter.sh/porter/pkg/tracing"
 	"github.com/pkg/errors"
 )
 
-var _ plugins.SecretsPlugin = &Store{}
+var _ plugins.SecretsProtocol = &Store{}
 
 const (
 	SECRET_FOLDER                          = "secrets"
 	FileModeSensitiveDirectory os.FileMode = 0700
 	FileModeSensitiveWritable  os.FileMode = 0600
 )
 
-// Config contains information needed for creatina a new store.
-type Config struct {
-	secretDir     string
-	debugLog      bool
-	porterHomeDir string
-}
-
-// NewConfig returns a new instance of Config.
-func NewConfig(debug bool, porterHomeDir string) Config {
-	return Config{
-		debugLog:      debug,
-		porterHomeDir: porterHomeDir,
-	}
-}
-
-// Valid checks if the configuration has been properly set.
-func (c Config) Valid() bool {
-	return c.secretDir != ""
-}
-
-// SetSecretDir configures the directory path for storing secrets.
-func (c *Config) SetSecretDir() (string, error) {
-	var err error
-	porterHomeDir := c.porterHomeDir
-	if porterHomeDir == "" {
-		porterCfg := config.New()
-		porterHomeDir, err = porterCfg.GetHomeDir()
-		if err != nil {
-			return "", errors.Wrap(err, "could not get user home directory")
-		}
-	}
-
-	c.secretDir = filepath.Join(porterHomeDir, SECRET_FOLDER)
-	return c.secretDir, nil
-}
-
 // Store implements an file system secrets store for testing and local
 // development.
 type Store struct {
-	config    Config
-	logger    hclog.Logger
-	hostStore cnabsecrets.Store
-	storage   aferox.Aferox
+	config    *config.Config
```

Comment:
I reworked this a bit to be consistent with the other plugins:

1. Each method is responsible for calling connect
2. Moved the creation of the secrets folder to connect
3. Passed a config struct in the constructor to simplify what we are passing (it already has context, filesystem, logs, etc)